code/game/objects/items/devices/radio/detpack.dm

```diff
@@ -0,0 +1,312 @@
+
+	if(do_mob(user, target, 30, BUSY_ICON_HOSTILE))
+		user.drop_held_item()
+		playsound(src.loc, 'sound/weapons/mine_armed.ogg', 25, 1)
+		var/location
+		location = target
+		forceMove(location)
+
+		message_admins("[key_name(user, user.client)](<A HREF='?_src_=holder;adminmoreinfo=\ref[user]'>?</A>) planted [src.name] on [target.name] at ([target.x],[target.y],[target.z] - <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[target.x];Y=[target.y];Z=[target.z]'>JMP</a>) with [timer] second fuse",0,1)
+		log_game("[key_name(user)] planted [src.name] on [target.name] at ([target.x],[target.y],[target.z]) with [timer] second fuse")
+
+		//target.overlays += image('icons/obj/items/assemblies.dmi', "plastic-explosive2")
+		user.visible_message("<span class='warning'>[user] plants [name] on [target]!</span>",
+		"<span class='warning'>You plant [name] on [target]! Timer set for [timer] seconds.</span>")
+
+		plant_target = target
+		if(istype(plant_target, /atom/movable))
+			var/atom/movable/T = plant_target
+			T.vis_contents += src
+			if(T.drag_delay < 3) //Anything with a fast drag delay we need to modify to avoid kamikazi tactics
+				target_drag_delay = T.drag_delay
+				T.drag_delay = 3
+		update_icon()
+
+
+/obj/item/device/radio/detpack/process()
+	//to_chat(world, "<font color='red'>DEBUG: Detpack Process.</font>")
+	if(plant_target == null || !plant_target.loc) //need a target to be attached to
+		//to_chat(world, "<font color='red'>DEBUG: Location mismatch.</font>")
+		processing_timers.Remove(src)
+		if(timer < DETPACK_TIMER_MIN) //reset to minimum 10 seconds; no 'cooking' with aborted detonations.
+			timer = DETPACK_TIMER_MIN
+		nullvars()
+		return
+	if(!on) //need to be active and armed.
+		processing_timers.Remove(src)
+		armed = FALSE
+		if(timer < DETPACK_TIMER_MIN) //reset to minimum 5 seconds; no 'cooking' with aborted detonations.
+			timer = DETPACK_TIMER_MIN
+		update_icon()
+		return
+	if(!armed)
+		if(timer < DETPACK_TIMER_MIN) //reset to minimum 5 seconds; no 'cooking' with aborted detonations.
+			timer = DETPACK_TIMER_MIN
+		//to_chat(world, "<font color='red'>DEBUG: Detpack Detonation Aborted.</font>")
+		processing_timers.Remove(src)
+		update_icon()
+		return
+	if(timer) //Timer is still counting down to armaggedon...
+		timer--
+		if(timer < 11)
+			playsound(src.loc, 'sound/weapons/mine_tripped.ogg', 160 + (timer-timer*2)*10, FALSE) //Gets louder as we count down to armaggedon
+		else
+			playsound(src.loc, 'sound/weapons/mine_tripped.ogg', 50, FALSE)
+		return
+	//Time to go boom
+	playsound(src.loc, 'sound/weapons/ring.ogg', 200, FALSE)
+	if(det_mode == TRUE) //If we're on demolition mode, big boom.
+		explosion(get_turf(plant_target), 2, 4, 5, 6)
+	else //if we're not, focused boom.
+		explosion(get_turf(plant_target), 1, 1, 2, 3)
+	if(plant_target)
+		plant_target.ex_act(1)
+		if(isobj(plant_target))
+			plant_target = null
+			if(!istype(plant_target,/obj/vehicle/multitile/root/cm_armored))

```

Comment:
I'd rather just delete everything except for stuff that obviously shouldn't be deleted which is accounted for.

Reply:
I started working on to replace unacidable var with FLAGS_1 stuff a while, so we get two different flags for INDESTRUCTIBLE and ACIDPROOF instead of a ballocks mix.
So feel free to discard this.